### DISCUSSION TITLE
matrix square root function only implemented on cpu?

### DISCUSSION THREAD

**Discussion by xinglong-li** [jax-ml/jax#11195] [Q&A]
I got the error message :

NotImplementedError: Schur decomposition is only implemented on the CPU backend.

when I try use the function: jax.scipy.linalg.sqatm  in computing the matrix square root on my GPU

**Comment by jakevdp** [comment]
That's true - the Schur decomposition was added in https://github.com/google/jax/pull/8033, but it only added a CPU implementation. Adding GPU or TPU support would involve writing a translation rule to target the appropriate GPU or TPU implementation.

**Reply by jakevdp** [reply]
I took a look through the cusolver docs (cusolver is what JAX lowers to for GPU translation rules for some other decompositions); it doesn't look like there's a schur implementation available, unfortunately.

**Reply by xinglong-li** [reply]
Got it.
Thanks for the clarification ;)
